### ISSUE TITLE
Error: No available formula with the name "homebrew/dupes/ncurses"

### ISSUE THREAD

**Issue by sjackman** [Homebrew/homebrew#50271]
I originally opened this issue over at https://github.com/Linuxbrew/linuxbrew/issues/938
@xu-cheng asked me to open a copy of that issue here.

Installing a formula that depends on `:python` before `homebrew/dupes` is tapped results in the error message `No available formula with the name "homebrew/dupes/ncurses"`. Note that on Linuxbrew `python` depends on `readline`, which depends on `homebrew/dupes/ncurses`. Installing a formula, which depends on `"python"` rather than `:python` works fine. e.g. `brew install gobject-introspection`. See below for examples.

@xu-cheng I believe this behaviour has changed recently, in the last week or so. Any thoughts on what changed?

### Fails

```
❯❯❯ brew install sphinx-doc
Error: No available formula with the name "homebrew/dupes/ncurses" 
Please tap it and then try again: brew tap homebrew/dupes
==> Searching for similarly named formulae...
Error: No similarly named formulae found.
==> Searching taps...
Error: No formulae found in taps.
```

### Succeeds

```
❯❯❯ brew install gobject-introspection
==> Tapping homebrew/dupes
Cloning into '/home/sjackman/.linuxbrew/Library/Taps/homebrew/homebrew-dupes'...
remote: Counting objects: 42, done.
remote: Compressing objects: 100% (42/42), done.
remote: Total 42 (delta 0), reused 5 (delta 0), pack-reused 0
Unpacking objects: 100% (42/42), done.
Checking connectivity... done.
Tapped 38 formulae (103 files, 105.5K)
==> Installing dependencies for gobject-introspection: xz, pkg-config, li
==> Installing gobject-introspection dependency: xz
==> Downloading https://linuxbrew.bintray.com/bottles/xz-5.2.2.x86_64_linux.bott
Already downloaded: /home/sjackman/.cache/Homebrew/xz-5.2.2.x86_64_linux.bottle.tar.gz
==> Pouring xz-5.2.2.x86_64_linux.bottle.tar.gz
…
```

**Comment by ilovezfs** [comment]
Is this a duplicate of https://github.com/Homebrew/homebrew/issues/50066 basically?

**Comment by xu-cheng** [comment]
@ilovezfs this is a different issue.

**Comment by ilovezfs** [comment]
Interesting. This is a regression introduced by f72d63fdd7320957ce60d6d2c51a44e170ee882a

I haven't checked yet whether it ever worked when the order of checksum and version is "correct" according to `brew audit --strict`, which is what broke it.

**Comment by ilovezfs** [comment]
@sjackman This seems not to be a regression. I can reproduce it with Homebrew from October 2014.

**Comment by ilovezfs** [comment]
@xu-cheng @sjackman https://github.com/Homebrew/homebrew/pull/50281 fixes this.

**Comment by sjackman** [comment]
Great. Thanks, @ilovezfs.

**Comment by sjackman** [comment]
My particular bug does not appear to have been fixed by this commit.
```sh
$ brew install node
Error: No available formula with the name "homebrew/dupes/ncurses" 
Please tap it and then try again: brew tap homebrew/dupes
==> Searching for similarly named formulae...
Error: No similarly named formulae found.
==> Searching taps...
Error: No formulae found in taps.
$ brew install sphinx-doc
Error: No available formula with the name "homebrew/dupes/ncurses" 
Please tap it and then try again: brew tap homebrew/dupes
==> Searching for similarly named formulae...
Error: No similarly named formulae found.
==> Searching taps...
Error: No formulae found in taps.
```

**Comment by ilovezfs** [comment]
@sjackman sure it is

```
openzfs@openzfs ~/.linuxbrew $ brew install node
Error: No available formula with the name "homebrew/dupes/ncurses" 
Please tap it and then try again: brew tap homebrew/dupes
==> Searching for similarly named formulae...
Error: No similarly named formulae found.
==> Searching taps...
Error: No formulae found in taps.
openzfs@openzfs ~/.linuxbrew $ brew pull https://github.com/Homebrew/homebrew/pull/50281
==> Fetching patch
######################################################################## 100.0%
==> Applying patch
Applying: FormulaInstaller: rescue tap unavail for reqs too
==> Patch closes issue #50281
==> Patch changed:
 Library/Homebrew/formula_installer.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
openzfs@openzfs ~/.linuxbrew $ vi Library/Homebrew/formula_installer.rb 
openzfs@openzfs ~/.linuxbrew $ brew install node
==> Tapping homebrew/dupes
Cloning into '/home/openzfs/.linuxbrew/Library/Taps/homebrew/homebrew-dupes'...
remote: Counting objects: 42, done.
remote: Compressing objects: 100% (42/42), done.
remote: Total 42 (delta 0), reused 3 (delta 0), pack-reused 0
Unpacking objects: 100% (42/42), done.
Checking connectivity... done.
Tapped 38 formulae (103 files, 106K)
^C
openzfs@openzfs ~/.linuxbrew $
```
